### ISSUE TITLE
bbPress 2 support

### ISSUE THREAD

**Issue by r-a-y** [r-a-y/bp-reply-by-email#24]
bbPress 2.2 was just released that has increased BuddyPress support, so RBE functionality would be ideal for RBE v1.1.

---

**What works currently in 1.0-RC1**
- Posting new forum topics via email for non-BuddyPress forums
- Posting new forum topics via email for BuddyPress groups (requires BP Group Email Subscription plugin)
- Posting forum replies via email for BuddyPress groups (requires BP Group Email Subscription plugin)

**What doesn't work**
<del>\* Posting forum replies via email for non-BuddyPress forums - this basically works, but is disabled at the moment. See below for more details.</del> - see commit b34d1be

**Remaining issues:**

<del>\* "Post link" in regular bbPress email notifications is inaccurate when replying by email - this is a bbPress bug and is holding up resolving this ticket.  Replying to a bbPress forum thread from a BP group works though because of GES.</del> - see commit b34d1be
<del>* Support multisite if RBE is network-activated (See #94)</del>
- Add RBE support for forum subscriptions

**Comment by boonebgorges** [comment]
Hi @r-a-y - Do you have a gut feeling about how hard it'd be to add support for bbPress's subscription emails? I have a client who'd like the functionality, and I could work on a PR, but it'd be helpful to know up front if there are going to be weird issues.

**Comment by r-a-y** [comment]
Are you talking about non-BP bbPress forums?  It should already work for topic subscriptions.  As noted in the OP, I haven't worked on forum subscriptions yet or tested if it works.  It might already work though.

At the moment, you have to use BuddyPress in tandem though.  You can't use this without BuddyPress due to how the plugin was originally created to support BP.  See #29.

**Comment by boonebgorges** [comment]
> Are you talking about non-BP bbPress forums?

No, I'm thinking of the following situation:
- Forums are in groups
- Running BPGES, but subscribed to Daily Digest or something other than All Mail
- User decides to subscribe to a single topic

> Update: Looks like it should be trivial to add forum subscription support. I'll add this in the next week or so.

Not knowing much about RBE's internals, my guess was that there'd be just a few moving parts. Seems like topic-based email hashes are already in place for All Email notifications, so it ought to be just a case of adding the proper `Reply-To` header for bbPress notifications. Like I said, I'm happy to dig deeper if you don't have the time, but if it really is trivial for you to do, I'd be glad to see it :-D

**Comment by boonebgorges** [comment]
> Have you tested your scenario to see what currently happens?

Yeah - RBE is not adding its `Reply-To` at all. (`get_reply_to_address()` is returning false because `$this->listener` is empty.) It looks like `BBP_RBE_Extension::extend_activity_listener()` should be setting up the listener, but maybe it's not doing it in time? I am not totally clear on what's happening here: it appears that the email is being sent using GES's email CPT (ie the HTML), even though it's being triggered by a bbPress subscription?

This site is on WP Engine, and I only have SFTP access, which makes it an absolute nightmare to debug. If you have any quick thoughts, that'd be helpful - otherwise I'll continue to chip away.

**Comment by r-a-y** [comment]
Hmm, I'm getting the bbPress email and the  `Reply-To` email address is being added... I'm also able to make a reply via email and it posts.  But subsequent subscription emails are not sent out, so there's a bug here.

> I am not totally clear on what's happening here: it appears that the email is being sent using GES's email CPT (ie the HTML), even though it's being triggered by a bbPress subscription?

Are you using thebrandonallen's plugin to send bbPress emails using the BuddyPress CPT and email template?
https://github.com/thebrandonallen/bp-emails-for-bbp

**Comment by r-a-y** [comment]
@boonebgorges - I've added some commits that should hopefully addresses the issue when a group user is subscribed to a topic with bbPress' bulit-in subscription functionality.

Give it a test and let me know if it works for you.

**Comment by boonebgorges** [comment]
Thank you! I will test and let you know.
